google/cloud/datastore/query.py

```diff
@@ -246,22 +395,49 @@ def add_filter(self, property_name, operator, value):
                      :class:`google.cloud.datastore.key.Key`
         :param value: The value to filter on.
 
+        :type filter: :class:`CompositeFilter`, :class:`PropertyFiler`
+        :param filter: A instance of a `BaseFilter`, either a `CompositeFilter` or `PropertyFilter`.
+
         :rtype: :class:`~google.cloud.datastore.query.Query`
         :returns: A query object.
 
         :raises: :class:`ValueError` if ``operation`` is not one of the
                  specified values, or if a filter names ``'__key__'`` but
                  passes an invalid value (a key is required).
         """
-        if self.OPERATORS.get(operator) is None:
-            error_message = 'Invalid expression: "%s"' % (operator,)
-            choices_message = "Please use one of: =, <, <=, >, >=, !=, IN, NOT_IN."
-            raise ValueError(error_message, choices_message)
+        if isinstance(property_name, PropertyFilter):
+            raise ValueError(
+                "PropertyFilter object must be passed using keyword argument 'filter'"
+            )
+        if isinstance(property_name, BaseCompositeFilter):
+            raise ValueError(
+                "Or and And objects must be passed using keyword argument 'filter'"
```

Comment:
The class names are `Or` and `And`. I quoted them as instead of uppercasing them hoping that reader will understand that it refers to the instance of `Or` or `And` class.